src/Microsoft.Xaml.Behaviors/InvokeCommandAction.cs

```diff
@@ -76,27 +112,60 @@ protected override void Invoke(object parameter)
 
                 if (command != null)
                 {
-                    if (command.CanExecute(this.CommandParameter))
+                    object commandParameter = this.CommandParameter;
+
+                    //if no CommandParameter has been provided, let's check the EventArgsParameterPath
+                    if (commandParameter == null && !string.IsNullOrWhiteSpace(this.EventArgsParameterPath))
                     {
-                        command.Execute(this.CommandParameter);
+                        commandParameter = GetEventArgsPropertyPathValue(parameter);
                     }
-                }
-                else
+
+                    //next let's see if an event args converter has been supplied
+                    if (commandParameter == null && this.EventArgsConverter != null)
+                    {
+                        commandParameter = this.EventArgsConverter.Convert(parameter, typeof(object), EventArgsConverterParameter, CultureInfo.CurrentCulture);
+                    }
+
+                    //last resort, let see if they want to force the event args to be passed as a parameter
+                    if (commandParameter == null && this.PassEventArgsToCommand)
+                    {
+                        commandParameter = parameter;
+                    }
+
+                    if (command.CanExecute(commandParameter))
+                    {
+                        command.Execute(commandParameter);
+                    }
+                } else
                 {
                     Debug.WriteLine(ExceptionStringTable.CommandDoesNotExistOnBehaviorWarningMessage, this.CommandName, this.AssociatedObject.GetType().Name);
                 }
             }
         }
 
+        private object GetEventArgsPropertyPathValue(object parameter)
+        {
+            object commandParameter;
+            object propertyValue = parameter;
+            string[] propertyPathParts = EventArgsParameterPath.Split('.');
+            foreach (string propertyPathPart in propertyPathParts)
+            {
+                PropertyInfo propInfo = propertyValue.GetType().GetProperty(propertyPathPart);
+                propertyValue = propInfo.GetValue(propertyValue, null);
+            }
+
+            commandParameter = propertyValue;
+            return commandParameter;
+        }
+
         private ICommand ResolveCommand()
         {
             ICommand command = null;
 
             if (this.Command != null)
             {
                 command = this.Command;
-            }
-            else if (this.AssociatedObject != null)
+            } else if (this.AssociatedObject != null)
```

Comment:
Looking through the codebase it seems the standard is for it to be on the new line. (another instance of this further up the file)

Reply:
Yeah, we have an editorconfig file that we need to update. I don't like this either. We should add an issue for an updated editorconfig file and set the new standards